interface/ceed.c

```diff
@@ -817,6 +817,62 @@ int CeedReference(Ceed ceed) {
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Computes the current memory usage of the work vectors in a `Ceed` context and prints to debug.abort
+
+  @param[in] ceed `Ceed` context
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref Developer
+**/
+int CeedGetWorkVectorMemoryUsage(Ceed ceed, CeedScalar *usage_mb) {
+  *usage_mb = 0.0;
+  if (ceed->work_vectors) {
+    for (CeedInt i = 0; i < ceed->work_vectors->num_vecs; i++) {
+      CeedSize vec_len;
+      CeedCall(CeedVectorGetLength(ceed->work_vectors->vecs[i], &vec_len));
+      *usage_mb += vec_len;
+    }
+    *usage_mb *= sizeof(CeedScalar) * 1e-6;
+    CeedDebug(ceed, "Resource {%s}: Work vectors memory usage: %" CeedInt_FMT " vectors, %g MB\n", ceed->resource, ceed->work_vectors->num_vecs,
+              *usage_mb);
+  }
+  return CEED_ERROR_SUCCESS;
+}
+
+/**
+  @brief Clear inactive work vectors in a `Ceed` context below a minimum length.
+
+  @param[in,out] ceed    `Ceed` context
+  @param[in]     min_len Minimum length of work vector to keep
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref Backend
+**/
+int CeedClearWorkVectors(Ceed ceed, CeedSize min_len) {
+  if (!ceed->work_vectors) return CEED_ERROR_SUCCESS;
+  for (CeedInt i = 0; i < ceed->work_vectors->num_vecs; i++) {
+    if (ceed->work_vectors->is_in_use[i]) continue;
+    CeedSize vec_len;
+    CeedCall(CeedVectorGetLength(ceed->work_vectors->vecs[i], &vec_len));
+    if (vec_len < min_len) {
+      ceed->ref_count += 2;  // Note: increase ref_count to prevent Ceed destructor from triggering again
+      CeedCall(CeedVectorDestroy(&ceed->work_vectors->vecs[i]));
+      ceed->ref_count -= 1;  // Note: restore ref_count
```

Comment:
```suggestion
      ceed->ref_count += 2;  // Note: increase ref_count to prevent Ceed destructor from triggering
      CeedCall(CeedVectorDestroy(&ceed->work_vectors->vecs[i]));
      ceed->ref_count -= 1;  // Note: restore ref_count
```
We probably don't need this, but its good to keep just in case. Minor wording change since this doesn't trigger via a call to `CeedDestroy()`